server/datastore/mysql/migrations/tables/20241022140321_AddStatusToHostMDM.go

```diff
@@ -0,0 +1,39 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20241022140321, Down_20241022140321)
+}
+
+func Up_20241022140321(tx *sql.Tx) error {
+	if !columnsExists(tx, "host_mdm", "enrollment_status", "created_at", "updated_at") {
+		if _, err := tx.Exec(`
+ALTER TABLE host_mdm
+ADD COLUMN enrollment_status ENUM('On (manual)', 'On (automatic)', 'Pending', 'Off') COLLATE utf8mb4_unicode_ci
+GENERATED ALWAYS AS (
+	CASE
+		WHEN is_server = 1 THEN NULL
+		WHEN enrolled = 1 AND installed_from_dep = 0 THEN 'On (manual)'
+		WHEN enrolled = 1 AND installed_from_dep = 1 THEN 'On (automatic)'
+		WHEN enrolled = 0 AND installed_from_dep = 1 THEN 'Pending'
+		WHEN enrolled = 0 AND installed_from_dep = 0 THEN 'Off'
+		ELSE NULL
+	END
+) VIRTUAL NULL,
```

Comment:
We discussed MySQL enums in the backend sync.

In this case, I moved this CASE statement to the DB so that I didn't have to duplicate its functionality in the code.

Reply:
It is a virtual table, so no extra DB storage is needed.

Reply:
Yes, it is an interesting pattern and nice to de-duplicate. One potential consideration is that we're embedding business logic/terminology even deeper into datastore layer. Of course, embedding it inside the queries also presented a similar concern, albeit closer to where the rest of the sausage is being made. I think it would make for an interesting discussion. It is something that's been on my mind for a while since this sort of business logic and terminology around Fleet changes more often than folks might expect.